static/app/views/releases/detail/overview/releaseComparisonChart/index.tsx

```diff
@@ -80,7 +95,94 @@ function ReleaseComparisonChart({
   reloading,
   errored,
   theme,
+  api,
+  organization,
 }: Props) {
+  const [eventsTotals, setEventsTotals] = useState<EventsTotals>(null);
+
+  const {
+    statsPeriod: period,
+    start,
+    end,
+    utc,
+  } = getReleaseParams({
+    location,
+    releaseBounds: getReleaseBounds(release),
+    defaultStatsPeriod: DEFAULT_STATS_PERIOD, // this will be removed once we get rid off legacy release details
+    allowEmptyPeriod: true,
+  });
+
+  useEffect(() => {
+    fetchEventsTotals();
+  }, [period, start, end, organization.slug, location]);
+
+  async function fetchEventsTotals() {
+    const url = `/organizations/${organization.slug}/eventsv2/`;
+    const commonQuery = {
+      environment: decodeList(location.query.environment),
+      project: decodeList(location.query.project),
+      start,
+      end,
+      ...(period ? {statsPeriod: period} : {}),
+    };
+
+    try {
+      const [
+        releaseTransactionTotals,
+        allTransactionTotals,
+        releaseErrorTotals,
+        allErrorTotals,
+      ] = await Promise.all([
```

Comment:
@markstory any chance this could be done in one request? (or at least less than four)

Reply:
Not with the current discover APIs. You could have an endpoint for release details that runs all the snuba queries. That would cut down some of your HTTP overhead.